src/http/index.ts

```diff
@@ -18,16 +18,22 @@ export interface HttpOptions {
   routesImporter?: RoutesImporter;
 
   // An array of valid express ApplicationRequestHandlers (middlewares) injected BEFORE loading routes
-  preRouteApplicationRequestHandlers?: any | [string, any][];
+  requestMiddleware?: any | [string, any][];
 
   // an array of valid express ApplicationRequestHandlers (middlewares) injected AFTER loading routes
-  postRouteApplicationRequestHandlers?: any | [string, any][];
+  errorMiddleware?: any | [string, any][];
+
+  // optional error hook function on application error
+  errorHook?: (error: any) => void;
+
+  // optional error logger which replaces console.error on application error
+  errorLogger?: (error: any) => void;
```

Comment:
not huge on 3 similar looking config options - that was the main thing to avoid, but at this point we just need to merge this and deal with cleanup later

Reply:
`errorHook` doesn't quite scream "do this when there is any error in the application" to me, but might just be me.

Something like `onApplicationError` is a bit more clear in my mind. Both these options happen around the same location and do very similar things, so that might be part of why it's not immediately clear, but without a bit of a restructure I don't have a better idea

Reply:
don't agree there, "hook" is a word used in a few systems to say "do this thing on XYZ"

Reply:
agreed there Pat, reverted back to nesting the http exception options, much cleaner this way